community/detectors/kubernetes_rce_via_open_access/src/main/java/com/google/tsunami/plugins/detectors/rce/kubernetes/RCEInKubernetesClusterWithOpenAccessDetector.java

```diff
@@ -0,0 +1,290 @@
+    this.payloadFormatString =
+        String.format(
+            Resources.toString(
+                Resources.getResource(this.getClass(), "payloadFormatString.json"), UTF_8),
+            RCE_POD_NAME,
+            "%s"); // Placeholder for the command payload
+  }
+
+  // This is the main entry point of VulnDetector.
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("RCEInKubernetesClusterWithOpenAccessDetector starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                // TODO: This filter doesn't trigger this detector.
+                // .filter(NetworkServiceUtils::isWebService)
+
+                // Check individual NetworkService whether it is vulnerable.
+                .filter(this::isServiceVulnerable)
+                // Build DetectionReport message for vulnerable services.
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  // Checks whether a given Kubernetes service is exposed and vulnerable.
+  private boolean isServiceVulnerable(NetworkService networkService) {
+
+    // Tell the PayloadGenerator what kind of vulnerability we are detecting
+    PayloadGeneratorConfig config =
+        PayloadGeneratorConfig.newBuilder()
+            .setVulnerabilityType(PayloadGeneratorConfig.VulnerabilityType.REFLECTIVE_RCE)
+            .setInterpretationEnvironment(
+                PayloadGeneratorConfig.InterpretationEnvironment.LINUX_SHELL)
+            .setExecutionEnvironment(
+                PayloadGeneratorConfig.ExecutionEnvironment.EXEC_INTERPRETATION_ENVIRONMENT)
+            .build();
+
+    // Pass in the config to get the actual payload from the generator.
+    // If the Tsunami callback server is configured, the generator will always try to return a
+    // callback-enabled payload.
+    Payload payload = this.payloadGenerator.generate(config);
+    String commandToInject = payload.getPayload();
+    String reqPayload = String.format(payloadFormatString, commandToInject);
+    boolean isPodCreated = false;
+    boolean isVulnerable = false;
+
+    // Create a new Kubernetes Pod with RCE payload in container command args
+    isPodCreated = createPod(networkService, RCE_POD_NAME, reqPayload);
+    if (!isPodCreated) {
+      logger.atInfo().log("Failed to create a pod. Not vulnerable.");
+      return false;
+    }
+
+    // Created a pod, appears vulnerable. Use callback for RCE confirmation if enabled
+    if (!payload.getPayloadAttributes().getUsesCallbackServer()) {
+      logger.atWarning().log("Pod created, appears vulnerable, but callback server is disabled.");
+      // NOTE: Set to false if only executed payloads/confirmed RCE cases should be reported
+      isVulnerable = true;
```

Comment:
How about we always report a vuln if pod creation is successful? We can set its severity to high. If callback server is enabled, we try RCE, if RCE exists, we raise its severity to CRITICAL. 

We can then monitor the finding quality, if we discover enough vulns, we can check if we can simplify the detection logic to only the pod creation part to keep the logic simple.

Reply:
Ok, I changed the plugin to always report the target as vulnerable when a pod gets successfully created with High severity. Confirming RCE with callback server raises the severity to Critical.